pkg/kubelet/network/plugins.go

```diff
@@ -38,6 +38,13 @@ type NetworkPlugin interface {
 	// before any other methods are called.
 	Init(host Host) error
 
+	// Tells the plugin to start any long-running operations; the plugin
+	// should not block this function
+	Run() error
+
+	// Indicates to the plugin that the pod CIDR changed, if controlled by kubelet
+	PodCIDRChanged(cidr string)
```

Comment:
@thockin after review, the only case where the PodCIDR changes is when it's retrieved from either flannel (with the flannelExpirmentalOverlay) or the master (with reconcileCIDR).  When that's the case, kubelet doens't know the PodCIDR when it initializes the network plugin so there needs to be a way to push that information down to the plugins.